### PR TITLE
New polynomial types

### PR DESCRIPTION
**Pull request by tcunis**
This PR introduces the polynomial data types

- `casos.PD` : Polynomials with constant coefficients using `DM`
- `casos.PS` : Polynomials with symbolic coefficients using `SX`

Also, the accompanying classes `casos.Indeterminates` and `casos.Sparsity` are introduced as well as linear operators between polynomials.

> [!IMPORTANT]
> With this PR, the **only** compatible version of CasADi is v3.6.0 or newer!

### CODE REVIEW

**Comment by tcunis** [comment]
Still todo:

- [ ] Add new types to read-me
- [ ] Use new types in demos

File: +casos/+package/+core/@AbstractOperator/AbstractOperator.m [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+            So = casos.Sparsity(0,0);
+            M = [];
+
+        elseif isa(M,'casos.package.core.AbstractOperator')
+            % copy or convert operator
+            Si = casos.Sparsity(M.sparsity_in);
+            So = casos.Sparsity(M.sparsity_out);
+            M = M.matrix;
+        % Note: Conversion from polynomial is ambiguous
+        % elseif isa(M,'casos.package.core.GenericPolynomial')
```
Comment (renatoloureiro) [inline +casos/+package/+core/@AbstractOperator/AbstractOperator.m:43]: Is this commented code unnecessary, still in 'To Do' or something else?

File: +casos/+package/+core/@AbstractOperator/AbstractOperator.m [hunk 2]
```diff
@@ -0,0 +1,258 @@
+classdef (Abstract) AbstractOperator < casos.package.core.PolynomialInterface
```
Comment (renatoloureiro) [inline +casos/+package/+core/@AbstractOperator/AbstractOperator.m:1]: No current demo or example in this repo tests out this new class. Can you provide a simple test file? 
(maybe add it to the branch for testing)

File: +casos/+package/+core/@AbstractOperator/AbstractOperator.m (hunk 1 above)
Comment (tcunis) [reply +casos/+package/+core/@AbstractOperator/AbstractOperator.m:43]: it's some mental note that we cannot have a conversion constructor from Polynomials to Operators.

File: +casos/+package/+core/@AbstractOperator/AbstractOperator.m (hunk 2 above)
Comment (tcunis) [reply +casos/+package/+core/@AbstractOperator/AbstractOperator.m:1]: both requests have been addressed.

**Review by renatoloureiro — APPROVED** [review]
1. All requested changes were completed and all questions were answered.
2. The demos and examples work without any noticeable issue.

**Pull request merged**